Gutenboarding: Font button style/size

sirreal opened the pull request:
#### Changes proposed in this Pull Request

This improves the font picker buttons overall to be closer to the designs in their size as well as fixes an issue with the font alignment.

Fixes #40206

#### Screens



##### Current
![Screen Shot 2020-03-19 at 17 07 42](https://user-images.githubusercontent.com/841763/77089697-7005d580-6a06-11ea-9809-8ee6ab4fff76.png)

##### Branch

![Screen Shot 2020-03-19 at 17 21 34](https://user-images.githubusercontent.com/841763/77089705-72682f80-6a06-11ea-837c-f6f6c3ace763.png)

##### 100% / design side-by-side

Design | Branch
--- | ---
![Screen Shot 2020-03-19 at 17 21 53](https://user-images.githubusercontent.com/841763/77089730-7a27d400-6a06-11ea-801e-55647503c6ee.png) | ![Screen Shot 2020-03-19 at 17 21 42](https://user-images.githubusercontent.com/841763/77089733-7ac06a80-6a06-11ea-8a3e-090b437fa29c.png)  


#### Testing instructions

Checkout and run locally. Visit `http://calypso.localhost:3000/gutenboarding` and proceed to this step.

sirreal commented:
> Looks like getting there. A couple of minor concerns:
> 
> 1. I notice the border shrinks when hovered. Is this expected, or something we can/should fix?
> 
> ![Kapture 2020-03-20 at 11 55 26](https://user-images.githubusercontent.com/1705499/77153869-477fe900-6aa3-11ea-9ddc-63bfbb16d859.gif)

Good spot, I'll fix that 👍 

> 1. The border seems to be a little more curved than the designs:
> 
> Branch	Design
> <img alt="Screenshot 2020-03-20 at 12 02 15 PM" width="209" src="https://user-images.githubusercontent.com/1705499/77153643-cd4f6480-6aa2-11ea-9070-12593e00b281.png">	<img alt="Screenshot 2020-03-20 at 12 01 25 PM" width="258" src="https://user-images.githubusercontent.com/1705499/77153660-d5a79f80-6aa2-11ea-9550-bcb8aa26930f.png">

I I tried to find a good middle ground between the designs and the styling of core components. I didn't touch the border radius from the core buttons.

@dubielzyk Curious for the design thoughts here, should we do our best to match designs at high fidelity, or is there a desire to match core component styling to some degree?

chriskmnds commented:
> Good spot, I'll fix that 👍

Quick check, there seems to be no hover effect in the prototype (https://hardcore-dubinsky-43a2f5.netlify.com/). Colors/borders don't change, only the selected:

![Kapture 2020-03-20 at 13 55 02](https://user-images.githubusercontent.com/1705499/77161567-ae58ce80-6ab2-11ea-9c95-e4faaa0057bd.gif)

sirreal commented:
Thanks for review, I've addressed the points, including the smaller border radius.

Let's leave the hover for now, I'm sure we'll want a hover style.

Updated screenshot:

![Screen Shot 2020-03-20 at 14 09 17](https://user-images.githubusercontent.com/841763/77166436-9124ff80-6ab4-11ea-8cbb-25327805c168.png)

simison commented on client/landing/gutenboarding/onboarding-block/style-preview/style.scss:
```diff
@@ -29,16 +29,24 @@
-	border: 1px solid var( --studio-gray-10 );
+.style-preview__font-option {
+	// Extra specificity to override core style
+	&.components-button {
```
Just a tiny nit; styles `min-height`, `box-shadow`, `justify-content` etc should be before this line as is a common pattern.  I think we've had a lint for it in the past...
